azurerm/internal/services/network/vpn_server_configuration_resource.go

```diff
@@ -237,14 +237,43 @@ func resourceArmVPNServerConfiguration() *schema.Resource {
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"address": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
 						},
 
 						"secret": {
-							Type:      schema.TypeString,
-							Required:  true,
-							Sensitive: true,
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+							Sensitive:    true,
+						},
+
+						"server": {
```

Comment:
Instead of adding a sub block here, can we just remove the max items on radius server, add score, and then if score is set, us the new property, and if score is not set & there is only 1 radius server use the old property?

Reply:
Agree, that's a better solution. I'll do the change.

Reply:
I forget to mention that after removing the constrain of max items on radius server, we need to lift client_root_certificate
and server_root_certificate in the radius server block to the outer scope, since these two settings are shared by all radius servers. If so, it will introduce a break-change. Now we're in the situation either introduce a break change or compromise a little a bit as what I did in the current PR. So before proceed,  I'd like to know your opinion on it.

Reply:
oh dang, didn't notice those! hmm

What we could do is keep `radius_server` as is and deprecated it then add a new block `radius` which would have those two properties and have a `servers` list inside. Then we can make them conflict so the user has to pick one or the other and we just use the one that is configured?

Reply:
I've done the change based on your suggestion.

Reply:
@katbyte  Since the waiting-response label still there, just ping you again in case you didn't notice my above reply. thanks.